spaceship/lib/spaceship/two_step_or_factor_client.rb

```diff
@@ -61,9 +61,17 @@ def handle_two_step_for_device(device_id)
 
       # Send token to server to get a valid session
       r = request(:post) do |req|
-        req.url("https://idmsa.apple.com/appleauth/auth/verify/device/#{device_id}/securitycode")
+        req.url("https://idmsa.apple.com/appleauth/auth/verify/phone/securitycode")
```

Comment:
So i think the original url was correct, but your body was wrong. 

In Xcodes - https://github.com/RobotsAndPencils/XcodesApp/blob/main/Xcodes/AppleAPI/Sources/AppleAPI/URLRequest%2BApple.swift#L68

Depending on if it's `sms` or `device` it'll send different bodies. 

for SMS: `verify/phone/securitycode`
```
  req.body = {
          "phoneNumber": {
            "id": device_id
          },
          "securityCode": {
            "code" => code.to_s
          },
          "mode": "sms"
```
for trusted device: ``verify/trusteddevice/securitycode`
```
  req.body = {
          "securityCode": {
            "code" => code.to_s
          }
```
That hasn't changed for a while and has been working, but maybe that has changed now 🤷

Reply:
Yeah yeah, the body definitely seemed wrong so not really sure how it didn't break sooner 🤷‍♂️ 

But... the new url/updated URL look correct though too, right? This is what proxyman showed me today 👇 

<img width="590" alt="Screenshot 2023-02-23 at 12 21 28 PM" src="https://user-images.githubusercontent.com/401294/220996386-40f6c8ca-8065-4f95-bb9d-e8d1d163d370.png">

Reply:
My guess is not many people use sms as their 2fa vs a trusted device

that Updated url looks correct for SMS 2FA. For trusted device it's: 

![image](https://user-images.githubusercontent.com/1119565/220998196-5075fe0a-61b1-48b1-9177-84259c8c364d.png)

Reply:
Ah yeah yeah, we have a separate path for trusted devices that uses that URL and body already 😊